drivers/serial/uart_ite_it8xxx2.c

```diff
@@ -0,0 +1,931 @@
+		*baud_rate_index = t_br_14400;
+		if (baud_rate != 14400) {
+			ret = STATUS_WARN_BAUDRATE;
+		}
+	} else if ((baud_rate <= 19200) && (baud_rate > 14400)) {
+		*baud_rate_index = t_br_19200;
+		if (baud_rate != 19200) {
+			ret = STATUS_WARN_BAUDRATE;
+		}
+	} else if ((baud_rate <= 38400) && (baud_rate > 19200)) {
+		*baud_rate_index = t_br_38400;
+		if (baud_rate != 38400) {
+			ret = STATUS_WARN_BAUDRATE;
+		}
+	} else if ((baud_rate <= 57600) && (baud_rate > 38400)) {
+		*baud_rate_index = t_br_57600;
+		if (baud_rate != 57600) {
+			ret = STATUS_WARN_BAUDRATE;
+		}
+	} else if ((baud_rate <= 115200) && (baud_rate > 57600)) {
+		*baud_rate_index = t_br_115200;
+		if (baud_rate != 115200) {
+			ret = STATUS_WARN_BAUDRATE;
+		}
+	} else if ((baud_rate <= 230400) && (baud_rate > 115200)) {
+		*baud_rate_index = t_br_230400;
+		if (baud_rate != 230400) {
+			ret = STATUS_WARN_BAUDRATE;
+		}
+	} else if (baud_rate > 230400) {
+		/* the max baud rate is 460800 */
+		*baud_rate_index = t_br_460800;
+		if (baud_rate != 460800) {
+			ret = STATUS_WARN_BAUDRATE;
+		}
+	}
+	return ret;
+}
+
+#define DEV_CFG(dev)					      \
+	((const struct uart_ite_it8xxx2_device_config *const) \
+	 (dev)->config->config_info)
+#define DEV_DATA(dev) \
+	((struct uart_ite_it8xxx2_dev_data_t *)(dev)->driver_data)
+struct uart_ite_it8xxx2_device_config {
+	uint32_t sys_clk_freq;
+};
+/** Device data structure */
+struct uart_ite_it8xxx2_dev_data_t {
+	uint32_t base;
+	uint32_t port;
+	uint32_t baud_rate;     /**< Baud rate */
+	uint32_t irq_no;
+	#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+
+	/**< cache of IIR read clear */
+	uint8_t iir_cache;
+
+	/**< Callback function pointer */
+	uart_irq_callback_user_data_t cb;
+
+	/**< Callback function arg */
+	void *cb_data;
+	#endif /* CONFIG_UART_INTERRUPT_DRIVEN */
+};
+
+static int uart_ite_it8xxx2_poll_in(struct device *dev, unsigned char *c)
+{
+	int ret = STATUS_SUCCESS;
+	struct uart_ite_it8xxx2_dev_data_t *const dev_data = DEV_DATA(dev);
+
+	ret = uart_getc(dev_data->port, c);
+	return ret;
+}
+
+static void uart_ite_it8xxx2_poll_out(struct device *dev, unsigned char c)
+{
+	struct uart_ite_it8xxx2_dev_data_t *const dev_data = DEV_DATA(dev);
+
+	uart_putc(dev_data->port, c);
+}
+
+static int uart_ite_it8xxx2_err_check(struct device *dev)
+{
+	struct uart_ite_it8xxx2_dev_data_t *const dev_data = DEV_DATA(dev);
+
+	return uart_is_xmit_empty(dev_data->port);
+}
+
+#if CONFIG_UART_INTERRUPT_DRIVEN
+/**
+ * @brief Fill FIFO with data
+ *
+ * @param dev UART device struct
+ * @param tx_data Data to transmit
+ * @param size Number of bytes to send
+ *
+ * @return Number of bytes sent
+ */
+static int uart_ite_it8xxx2_fifo_fill(
+	struct device *dev, const uint8_t *tx_data, int size)
+{
+	int i, val = 0;
+	unsigned char ucTemp;
+	struct uart_ite_it8xxx2_dev_data_t *const dev_data = DEV_DATA(dev);
+	volatile struct T_ITEUART *pComPort =
+		(volatile struct T_ITEUART *)dev_data->base;
+
+#if DEF_UART2_ENABLE_CTS_RTS
```

Comment:
I don't know what DEF_UART2_ENABLE_CTS_RTS is, but I think this should use if() instead of #if. It is better for build coverage and easier to read. The resulting compiler output should be the same. Please fix globally